theano/gpuarray/c_code/dnn_fwd.c

```diff
@@ -231,7 +231,8 @@ APPLY_SPECIFIC(conv_fwd)(PyGpuArrayObject *input, PyGpuArrayObject *kerns,
           return -1;
         }
         // set the 'tensor math ok' flag
-        c_set_math_type_for_conv(desc, CUDNN_TENSOR_OP_MATH);
+        if (input->ga.typecode == GA_HALF)
```

Comment:
I understand that enabling that when input are float32 reduce the precision as you wrote in the description. But if it is enabled, does it speed up computation?

One option would be to cast the input to half in the graph before doing computation in float16, vs letting cudnn read in float32 and compute in float16. Is there one better?

At GTC, I recall that they kept the weights in float32, but casted it to float16 for computation. But I don't recall them telling if they would do a cast and call cudnn on the float16 result or let cudnn read float32 and convert it itself to float16.

Reply:
Thanks for the input, @nouiz !
I have raised the question with CUDNN team - it's understood that loss of precision should not be allowed by default - however, it may be a useful option to test HMMA speed without changing any code.